src/components/Link/Link.tsx

```diff
@@ -190,20 +197,27 @@ export const Link = chakra(
       const childrenToClone: any = children[0] ? children[0] : children;
       const childProps = childrenToClone.props;
       return (
-        <Box as="span" __css={styles} {...rest}>
-          {React.cloneElement(
-            childrenToClone,
-            {
-              className,
-              ...linkProps,
-              ...childProps,
-              ref,
-              rel,
-              target: internalTarget,
-            },
-            [childrenToClone.props.children]
+        <>
+          {screenreaderOnlyText && (
+            <Box as="span" __css={styles.screenreaderOnly}>
+              {screenreaderOnlyText}
+            </Box>
```

Comment:
Same thing here regarding the positioning.

Reply:
Yes. Similar issue happens here that is mentioned for the `Button` component above. The children are not always "text -> Icon".